pkg/providers/imagefamily/azlinux3.go

```diff
@@ -0,0 +1,135 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package imagefamily
+
+import (
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1beta1"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily/bootstrap"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily/customscriptsbootstrap"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/launchtemplate/parameters"
+
+	karpv1 "sigs.k8s.io/karpenter/pkg/apis/v1"
+	"sigs.k8s.io/karpenter/pkg/cloudprovider"
+	"sigs.k8s.io/karpenter/pkg/scheduling"
+)
+
+const (
+	AzureLinux3Gen2ImageDefinition    = "V3gen2"
+	AzureLinux3Gen1ImageDefinition    = "V3"
+	AzureLinux3Gen2ArmImageDefinition = "V3gen2arm64"
+)
```

Comment:
Looks correct to me based on our current support, and the previous real world data from the NodeImageVersions API. However, any need to double check with Node SIG?

Reply:
They match what AgentBaker has, e.g. https://github.com/Azure/AgentBaker/blob/c03652307949f9259dfc84aca858afdaf788e692/pkg/agent/datamodel/sig_config.go#L768